### ISSUE TITLE
Provide no-token diarization alternative

### ISSUE THREAD

**Issue by DakotaIrsik** [irsiksoftware/TalkSmith#7]
## Overview
Implement alternative diarization using embeddings + spectral clustering (e.g., Resemblyzer) for environments without HuggingFace token access.

## Current Status
- ✅ `pipeline/transcribe_fw.py` - Core transcription exists
- ❌ `pipeline/diarize_whisperx.py` - Primary diarization (requires HF token)
- ❌ `pipeline/diarize_alt.py` - Token-free alternative (NOT YET IMPLEMENTED)

## Why This Matters
- pyannote.audio (used by WhisperX) requires HF token with accepted terms
- Some deployments cannot use HF tokens (air-gapped, privacy-restricted)
- Alternative needed for broader compatibility

### Option 1: Resemblyzer + Spectral Clustering
```python
# Use speaker embeddings without external API
from resemblyzer import VoiceEncoder, preprocess_wav
from sklearn.cluster import SpectralClustering

# Extract embeddings from segments
# Cluster by speaker similarity
# Assign speaker labels
```

### Option 2: Simple-Diarizer
```python
# Lightweight alternative using Resemblyzer
import diarizer
# Simplified API without HF dependencies
```

### Option 3: Speechbrain
```python
# Open-source toolkit with embedding models
from speechbrain.pretrained import EncoderClassifier
# Pre-trained models available without tokens
```

## Acceptance Criteria
- [ ] `pipeline/diarize_alt.py` implements token-free diarization
- [ ] CLI interface: `python pipeline/diarize_alt.py <audio> -o segments.json`
- [ ] Output format matches WhisperX output (start, end, speaker, text)
- [ ] Quality comparison documented in `docs/diarization.md`
- [ ] Performance benchmarks vs. pyannote
- [ ] Installation instructions (no HF token required)
- [ ] README updated with token-free option

## Quality Trade-offs
Document in `docs/diarization.md`:
- Accuracy differences vs. pyannote
- Speaker count limitations
- Best use cases for each approach
- When to use token-free vs. HF-token method

## Priority
**P2** - Important for deployment flexibility, but not critical for initial functionality

## Dependencies
- Should work with existing `FasterWhisperTranscriber` output
- Optional: Can be integrated with WhisperX alignment (without diarization)

## Testing
```bash
# Test token-free diarization
python pipeline/diarize_alt.py data/samples/test-audio.wav -o output.json

# Compare with WhisperX
pytest tests/test_diarization_comparison.py -v
```

**Comment by DakotaIrsik** [comment]
🔓 **Unblocked** - Issue #5 (faster-whisper implementation) is now complete. However, this also depends on #6 (WhisperX). Once #6 is done, you can implement the alternative diarization.
